### PR TITLE
provider/aws: Add JSON validation to the aws_elasticsearch_domain resource.

### PR DESCRIPTION
**Pull request by kwilczynski**
This commit adds support for new helper function which is used to
normalise and validate JSON string.

Signed-off-by: Krzysztof Wilczynski <krzysztof.wilczynski@linux.com>

### CODE REVIEW

**Comment by kwilczynski** [comment]
Part of work which began in #8028.

**Comment by kwilczynski** [comment]
Acceptance test is passing:

```
$ make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSElasticSearchDomain_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/09/17 23:00:25 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSElasticSearchDomain_ -timeout 120m
=== RUN   TestAccAWSElasticSearchDomain_basic
--- PASS: TestAccAWSElasticSearchDomain_basic (2162.51s)
=== RUN   TestAccAWSElasticSearchDomain_v23
--- PASS: TestAccAWSElasticSearchDomain_v23 (1980.72s)
=== RUN   TestAccAWSElasticSearchDomain_complex
--- PASS: TestAccAWSElasticSearchDomain_complex (2074.76s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/aws   	6218.016s
```

File: builtin/providers/aws/resource_aws_elasticsearch_domain.go [hunk 1]
```diff
@@ -269,7 +270,8 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 	ds := out.DomainStatus
 
 	if ds.AccessPolicies != nil && *ds.AccessPolicies != "" {
-		d.Set("access_policies", normalizeJson(*ds.AccessPolicies))
+		policies, _ := normalizeJsonString(*ds.AccessPolicies)
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_elasticsearch_domain.go]: We need to check for errors ere ;)

**Comment by stack72** [comment]
LGTM! Thanks for the work here :)

**Pull request merged**